sdks/python/apache_beam/pipeline.py

```diff
@@ -157,6 +157,146 @@ def _root_transform(self):
     """Returns the root transform of the transform stack."""
     return self.transforms_stack[0]
 
+  def _remove_labels_recursively(self, applied_transform):
+    for part in applied_transform.parts:
+      if part.full_label in self.applied_labels:
+        self.applied_labels.remove(part.full_label)
+      if part.parts:
+        for part2 in part.parts:
+          self._remove_labels_recursively(part2)
+
+  def _replace(self, override):
+
+    assert isinstance(override, PTransformOverride)
+    matcher = override.get_matcher()
+
+    output_map = {}
+    output_replacements = {}
+    input_replacements = {}
+
+    class TransformUpdater(PipelineVisitor): # pylint: disable=used-before-assignment
+      """"A visitor that replaces the matching PTransforms."""
+
+      def __init__(self, pipeline, applied_labels):
+        self.pipeline = pipeline
+        self._applied_labels = applied_labels
+
+      def _replace_if_needed(self, transform_node):
+        if matcher.match(transform_node):
+          replacement_transform = override.get_replacement_transform(
+              transform_node.transform)
+          inputs = transform_node.inputs
+          # TODO:  Support replacing PTransforms with multiple inputs.
+          if len(inputs) > 1:
+            raise NotImplementedError(
+                'PTransform overriding is only supported for PTransforms that '
+                'have a single input. Tried to replace %r that has %d inputs',
+                transform_node, len(inputs))
+          assert len(inputs) == 1
```

Comment:
Remove this assert, it's redundant with the if above (or could be, if it was equality rather than greater than).